python/ray/_private/utils.py

```diff
@@ -1624,3 +1625,15 @@ def is_in_test():
             for env_var in ("PYTEST_CURRENT_TEST", "BUILDKITE")
         )
     return in_test
+
+
+def is_ipv6_address(ip):
```

Comment:
This method strips leading `[]` but won’t accept link-local addresses with a zone ID (e.g. "fe80::1%eth0"). If some environment use such addresses, you’ll need to extend the util to handle `%`-notation correctly.

Also, this method looks like a good candidate to move to `python/ray/_common` instead of `_private`. Let's also add comprehensive unit tests for this method in `python/ray/_common/tests/test_utils.py`